src/login/logind-seat-dbus.c

```diff
@@ -400,6 +387,10 @@ int seat_node_enumerator(sd_bus *bus, const char *path, void *userdata, char ***
         assert(path);
         assert(nodes);
 
+        r = strv_extend(&l, "/org/freedesktop/login1/seat/self");
+        if (r < 0)
+                return r;
```

Comment:
no I define them so they _do_ exist :).

It's more magic if the objects disappear and reappear depending who you are.  If you want to find out why it didn't exist, you had to call the method to query your session.  But in that case, there was no point the client using session guessing in the first place.

I could have spelled that out much better in the commit messages.  This PR went through a few rounds :) and I didn't nail all the messages.

Reply:
There's a rough analogy with `/dev/stdin`.

Also, it's hard to distinguish whether an object doesn't exist dynamically, or whether you're talking to an older version that didn't implement that part of the API.  Pure non-existence is used to detect the latter, it's called out in DBus API guidelines.

Reply:
hmm, i am not sure i follow. what do you mean by saying the "do exist"? what object is actually on that object path if you are calling from a session-less process? I mean, am I missing something here, you don't actually synthesize a proper object here, do you? you just say that there is one, but if one would then ask for it there's actually nothing there...

Note that the "self" object path is just an alias name anyway... the set of objects doesn't really change, it#s just an alias that becomes available depending on who you are.

I am not sure how i feel if i browse objects with d-feed or busctl and i see an object listed, but when i want to figure out its details it vanishes...

Reply:
In terms of d-feet/busctl, that's exactly what I'd want.  If I'm looking into this, I want to be able to see the specific error message.  If I can hypothetically click on the path in d-feet and get the specific error message it a) confirms that there's specific code to handle this path; b) gestures towards the conditions that let the alias work.  This is more specifically helpful than having d-feet show an object I can interact with entirely sucessfully, but busctl showing it as non-existent and returning a generic error message.  (d-feet doesn't run as a service on F26.  But I'm running busctl from gnome-terminal-server.)

Yes, this is a very fine point to argue.  But I don't believe it would be acceptable to return non-zero from session_object_find(), if we're not including the corresponding path in the result of session_node_emumerator().  That's what I mean by the object existing but potentially returning an error.

The alternative that avoids arguing fine points is to consider all the `/self` paths as inappropriate if you want to report errors, probably not used anywhere, and sub-optimal for debugging.  [Here's an outline for user/self](https://github.com/sourcejedi/systemd/commit/2151ece3f82590816d60f26ef216792538d9e09f).

TODO: test this branch for the behaviour of each `/self` object.  I just noticed the current code is already returning non-zero from _find() for one path, which it denied knowing anything about -

```
# busctl tree introspect org.freedesktop.login1 | grep self
#

# busctl introspect org.freedesktop.login1 /org/freedesktop/login1/session/self
Failed to introspect object /org/freedesktop/login1/session/self of service org.freedesktop.login1: No such device or address

# busctl introspect org.freedesktop.login1 /org/freedesktop/login1/user/self
Failed to introspect object /org/freedesktop/login1/user/self of service org.freedesktop.login1: Unknown object '/org/freedesktop/login1/user/self'.
```

Reply:
Ah, actual behaviour of d-feet with this code is to show the error when you open the `login1` name, and not show `session/self`.  So I can't get away with it.